### DISCUSSION TITLE
Seek audio

### DISCUSSION THREAD

**Discussion by emretural** [pschatzmann/arduino-audio-tools#353] [Q&A] [answered]
Hey there,

Thanks for the amazing work !
I'm trying to play audio files from sd card using i2s according to commands coming from bluetooth and it works like a charm
But i also need to seek audio file. Like start playing "this" audio from "this" second
I couldn't find a function for that. I was wondering if thats possible

Thanks in advance everyone

**Comment by pschatzmann** [comment]
You could implement a simple stream filter that ignores the first n samples and provide a method which translates time to number samples with the help of the sample rate...

Another approach, that you could implement in your sketch, would be that you start with a AudioCopy with the destination to NullStream. You sum up the result of the copy() calls, and after you get valid data you copy to the real destination...

**Reply by luke2023** [reply]
great idea, but it requires a lot of time if I have a big file right ?

**Comment by pschatzmann** [comment]
It is worth a trial: it is driven by the decoding speed which should be faster than the playing speed. If you use wav files, you will not even notice any delay...

**Reply by luke2023** [reply]
Nice!Do  you have any mp3 beta or examples for that ?Thank you for your helping!``

**Comment by pschatzmann** [comment, marked as answer]
I just added the [TimeInfoStream](https://github.com/pschatzmann/arduino-audio-tools/blob/main/src/Experiments/TimeInfoStream.h) in Experiments: I suggest that you have a look at the available methods and run some tests.
You could wrap e.g. an I2SStream class:
```
#include "Experiments/TimeInfoStream.h"

I2SStream i2s;
TimeInfoStream timedOut(i2s);
```
